Library/Homebrew/os/mac/version.rb

```diff
@@ -20,35 +23,44 @@ class Version < ::Version
         yosemite:    "10.10",
       }.freeze
 
+      sig { params(sym: Symbol).returns(T.attached_class) }
       def self.from_symbol(sym)
         str = SYMBOLS.fetch(sym) { raise MacOSVersionError, sym }
         new(str)
       end
 
+      sig { params(value: T.nilable(String)).void }
       def initialize(value)
-        super(value)
+        raise MacOSVersionError, value unless /\A1\d+(?:\.\d+){0,2}\Z/.match?(value)
 
-        raise MacOSVersionError, value unless value.match?(/\A1\d+(?:\.\d+){0,2}\Z/)
+        super(value)
 
         @comparison_cache = {}
       end
 
       def <=>(other)
         @comparison_cache.fetch(other) do
-          v = SYMBOLS.fetch(other) { other.to_s }
-          @comparison_cache[other] = super(::Version.new(v))
+          if SYMBOLS.key?(other) && to_sym == other
+            0
+          else
+            v = SYMBOLS.fetch(other) { other.to_s }
+            @comparison_cache[other] = super(::Version.new(v))
+          end
         end
       end
 
+      sig { returns(Symbol) }
       def to_sym
-        SYMBOLS.invert.fetch(@version, :dunno)
+        @to_sym ||= SYMBOLS.invert.fetch((major >= 11 ? major : major_minor).to_s, :dunno)
```

Comment:
`(major >= 11 ? major : major_minor)` 

might be nice to split this into another helper method?

Reply:
Couldn't think of a good name so I left it as-is.

Reply:
`major_macos` or something?